viewer/packages/tools/src/Toolbar/ToolbarTool.ts

```diff
@@ -0,0 +1,143 @@
+ */
+export class ToolbarTool extends Cognite3DViewerToolBase {
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _model: CogniteModelBase;
+  private readonly _toolbar: Toolbar;
+  private axisTool: AxisViewTool | undefined;
+  private cameraControlOption: CameraControlsOptions;
+
+  constructor(viewer: Cognite3DViewer, model: CogniteModelBase) {
+    super();
+
+    this._viewer = viewer;
+    this._model = model;
+    this._toolbar = new Toolbar(this._viewer);
+    this.cameraControlOption = {
+      onClickTargetChange: false,
+      mouseWheelAction: 'zoomPastCursor'
+    };
+  }
+
+  /**
+   * Add a Icon button into the Toolbar container
+   * @param toolTip Tooltip for the Icon button
+   * @param backgroundImageUri background image to be placed onto icon
+   * @param isToggle Is the icon button used as toggle
+   * @param onClick Click event callback function which will be used to perform custom functionlity of the user
+   */
+  public addToolbarItem(toolTip: string, backgroundImageUri: string, isToggle: boolean, onClick: () => void): void {
+    this._toolbar.addToolbarItem(toolTip, backgroundImageUri, isToggle, onClick);
+  }
+
+  /**
+   * Set the position of the toolbar container
+   * @param position ToolbarPosition value such as Top, Bottom, Left, Right within the canvas
+   */
+  public setPosition(position: ToolbarPosition): void {
+    this._toolbar.setPosition(position);
+  }
+
+  /**
+   * Add Enable or Disable Axis view tool to the Toolbar
+   */
+  public addAxisToolToggle(): void {
+    this.addToolbarItem('Axis Tool', iconSet14, true, this.axisView);
+  }
+
+  /**
+   * Add Save the screenshot of the canvas been rendered to the Toolbar
+   */
+  public addTakeScreenshotTool(): void {
+    this.addToolbarItem('Take Screenshot', iconSet12, false, this.screenShot);
+  }
+
+  /**
+   * Adds Toggle enabling/disabling the Camera target on mouse click to the Toolbar
+   */
+  public addCameraTargetOnClickToggle(): void {
+    this.addToolbarItem('Enable/Disable Camera Target on Click', iconSet19, true, this.changeCameraTargetOnClick);
+  }
+
+  /**
+   * Adds Toggle camera for "Zoom Past Cursor" and "Zoom To Cursor" in to Toolbar
+   */
+  public addZoomPastToCursorToggle(): void {
+    this.addToolbarItem('Toggle Zoom past/Zoom to Cursor', iconSet30, true, this.toggleCameraZoomPastToCursor);
+  }
+
+  /**
+   * Adds Fit Camera to Model to Toolbar
+   */
+  public addFitCameraToModel(): void {
+    this.addToolbarItem('Fit Camera to Model', iconSet32, false, this.fitCameraToModel);
+  }
+
+  /**
+   * Enable or Disable Axis view tool
+   */
+  private readonly axisView = (): void => {
+    if (this.axisTool === undefined) {
+      this.axisTool = new AxisViewTool(this._viewer);
+    } else {
+      this.axisTool.dispose();
+      this.axisTool = undefined;
+    }
+  };
+
+  /**
+   * Save the screenshot of the canvas been rendered
+   */
+  private readonly screenShot = async (): Promise<void> => {
```

Comment:
Replace all "lambda style functions" with "plain functions", e.g. 
`private readonly screenShot = async (): Promise<void> => {`
to
 `private async screenShot(): Promise<void> {`